stackslib/src/chainstate/nakamoto/test_signers.rs

```diff
@@ -147,14 +252,66 @@ impl TestSigners {
         let signature = sig_aggregator
             .sign(msg.as_slice(), &nonces, &sig_shares, &key_ids)
             .expect("aggregator sig failed");
+        ThresholdSignature(signature)
+    }
 
-        test_debug!(
-            "Signed Nakamoto block {} with {} (rc {})",
-            block.block_id(),
-            &self.aggregate_public_key,
-            cycle
-        );
-        block.header.signer_signature = ThresholdSignature(signature);
+    /// Reorder a list of signatures to match the order of the reward set.
+    pub fn reorder_signatures(
+        &self,
+        signatures: Vec<MessageSignature>,
+        reward_set: &RewardSet,
+    ) -> Vec<MessageSignature> {
+        let test_signer_keys = &self
+            .signer_keys
+            .iter()
+            .cloned()
+            .map(|key| Secp256k1PublicKey::from_private(&key).to_bytes_compressed())
+            .collect::<Vec<_>>();
+
+        let reward_set_keys = &reward_set
+            .clone()
+            .signers
+            .unwrap()
+            .iter()
+            .map(|s| s.signing_key.to_vec())
+            .collect::<Vec<_>>();
+
+        let signature_keys_map = test_signer_keys
+            .iter()
+            .cloned()
+            .zip(signatures.iter().cloned())
+            .collect::<HashMap<_, _>>();
+
+        let mut reordered_signatures = Vec::with_capacity(reward_set_keys.len());
+
+        let mut missing_keys = 0;
+
+        for key in reward_set_keys {
+            if let Some(signature) = signature_keys_map.get(key) {
+                reordered_signatures.push(signature.clone());
+            } else {
+                missing_keys += 1;
+            }
+        }
+        if missing_keys > 0 {
+            warn!(
+                "TestSigners: {} keys are in the reward set but not in signer_keys",
+                missing_keys
+            );
+        }
+
+        reordered_signatures
+    }
+
+    // Sort [`Self::signer_keys`] by their compressed public key
+    pub fn sorted_signer_keys(&self) -> Vec<Secp256k1PrivateKey> {
```

Comment:
Why does this method need to exist?  I thought blocks were signed by their respective signers in reward-set order?

Reply:
You could do reward-set order (for each in reward set, lookup if test signer has that key, sign), but `TestSigners` is kind of doing the opposite (for each key in signer_keys, sign, then reorder/remove based on reward set). I can see how that's messy though, I'll do it the other way. Also, this specific function isn't even used. I probably used it at some point but I'll remove it

Reply:
I refactored this method to do the (for each key in reward_set, sign if TestSigner has that key) approach. LMK if you think this needs more work